crates/sui-mvr-graphql-e2e-tests/tests/tests.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(unused_imports)]
+#![allow(unused_variables)]
+use async_trait::async_trait;
+use std::{path::Path, sync::Arc, time::Duration};
+use sui_mvr_graphql_rpc::test_infra::cluster::{serve_executor, ExecutorCluster};
+use sui_transactional_test_runner::{
+    args::SuiInitArgs,
+    create_adapter,
+    offchain_state::{OffchainStateReader, TestResponse},
+    run_tasks_with_adapter,
+    test_adapter::{SuiTestAdapter, PRE_COMPILED},
+};
+
+pub struct OffchainReaderForAdapter {
+    cluster: Arc<ExecutorCluster>,
+}
+
+#[async_trait]
+impl OffchainStateReader for OffchainReaderForAdapter {
+    async fn wait_for_objects_snapshot_catchup(&self, base_timeout: Duration) {
+        self.cluster
+            .wait_for_objects_snapshot_catchup(base_timeout)
+            .await
+    }
+
+    async fn wait_for_checkpoint_catchup(&self, checkpoint: u64, base_timeout: Duration) {
+        self.cluster
+            .wait_for_checkpoint_catchup(checkpoint, base_timeout)
+            .await
+    }
+
+    async fn wait_for_pruned_checkpoint(&self, checkpoint: u64, base_timeout: Duration) {
+        self.cluster
+            .wait_for_checkpoint_pruned(checkpoint, base_timeout)
+            .await
+    }
+
+    async fn execute_graphql(
+        &self,
+        query: String,
+        show_usage: bool,
+    ) -> Result<TestResponse, anyhow::Error> {
+        let result = self
+            .cluster
+            .graphql_client
+            .execute_to_graphql(query, show_usage, vec![], vec![])
+            .await?;
+
+        Ok(TestResponse {
+            http_headers: Some(result.http_headers_without_date()),
+            response_body: result.response_body_json_pretty(),
+            service_version: result.graphql_version().ok(),
+        })
+    }
+}
+
+datatest_stable::harness!(run_test, "tests", r"stable/.*\.move$");
+
+#[cfg_attr(not(msim), tokio::main)]
+#[cfg_attr(msim, msim::main)]
+async fn run_test(path: &Path) -> Result<(), Box<dyn std::error::Error>> {
+    telemetry_subscribers::init_for_testing();
+    if !cfg!(msim) {
+        // start the adapter first to start the executor (simulacrum)
+        let (output, mut adapter) =
+            create_adapter::<SuiTestAdapter>(path, Some(Arc::new(PRE_COMPILED.clone()))).await?;
+
+        // In another crate like `sui-mvr-graphql-e2e-tests`, this would be the place to translate
+        // from `offchain_config` to something compatible with the indexer and graphql flavor of
+        // choice.
```

Comment:
nit: this comment needs updating since we are now in `sui-mvr-graphql-e2e-tests`!